test/topology_custom/test_conflicting_keys_read_repair.py

```diff
@@ -0,0 +1,85 @@
+#
+# Copyright (C) 2024-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+import logging
+import pytest
+import asyncio
+import time
+
+from cassandra import ConsistencyLevel  # type: ignore
+from cassandra.query import SimpleStatement  # type: ignore
+from test.pylib.manager_client import ManagerClient
+from test.pylib.util import wait_for_cql_and_get_hosts
+
+
+logger = logging.getLogger(__name__)
+
+
+@pytest.mark.xfail(reason="https://github.com/scylladb/scylladb/issues/19101")
+@pytest.mark.asyncio
+async def test_read_repair_with_conflicting_hash_keys(request: pytest.FixtureRequest, manager: ManagerClient) -> None:
+    """
+    Test that conflicting hash keys are handled correctly during read repair.
+    Issue https://github.com/scylladb/scylladb/issues/19101
+
+    1. Create a new cluster with 3 nodes.
+    2. Create a keyspace and a table with replication factor = 3.
+    3. Stop one of the nodes.
+    4. Add with 2 rows that have primary keys causing a hash collision.
+    5. Start the offline node.
+    6. Run a SELECT query with ALL consistency level, forcing reading from all 3 nodes.
+       The node that's been offline will not have a value, causing a read repair.
+       Since difference calculation logic is using a token for it's hashmap key and the
+       token is value is the same for both keys, this causes an incorrect diff calculation
+       and propagation to the node that was offline.
+    7. Run the same SELECT query with ALL consistency level, forcing reading from all 3 nodes.
+       now there is also a conflict, since the node that was reset got an incorrect value as a
+       result of and prev step read repair. This value is newer than others, thus the resulting
+       value will be incorrect in case the diff calculation algorithm is using a token as a key.
+
+    """
+    logger.info("Creating a new cluster")
+    for i in range(3):
+        await manager.server_add()
+
+    cql = manager.get_cql()
+    srvs = await manager.running_servers()
+    await wait_for_cql_and_get_hosts(cql, srvs, time.time() + 60)
```

Comment:
You can replace all these lines with
```python
srvs = await manager.servers_add(3)
cql, _ = await manager.get_ready_cql(srvs)
```

It will also make the test faster because `await manager.servers.add(3)` boots 3 servers concurrently. Calling `await manager.server_add()` multiple times still makes sense sometimes, for example, it ensures that the first node will be the group 0 leader.

Reply:
Thanks, changed.